modules/network-annotation-processor/src/main/java/org/apache/ignite/internal/network/processor/EnumMethodsGenerator.java

```diff
@@ -0,0 +1,182 @@
+import javax.lang.model.type.TypeMirror;
+import org.apache.ignite.internal.network.annotations.Transferable;
+import org.apache.ignite.internal.network.annotations.TransferableEnum;
+import org.apache.ignite.internal.network.annotations.Transient;
+
+/**
+ * Generates classes for {@link Enum} that are present in network messages (with {@link Transferable}) with utility methods.
+ *
+ * <p>{@link Enum} are expected to implement {@link TransferableEnum}.</p>
+ *
+ * <p>Name of the generated class will be the class name + {@code TransferableUtils}, for example for the enum {@code SomeEnum} the class
+ * name will be {@code SomeEnumTransferableUtils}.</p>
+ *
+ * <p>Generated methods:</p>
+ * <ul>
+ *     <li>{@code public static SomeEnum formTransferableId(int transferableId) throws IllegalInitArgumentException} - which will return
+ *     the enumeration constant by its {@link TransferableEnum#transferableId} and throw {@link IllegalArgumentException} if it does not
+ *     find it.</li>
+ * </ul>
+ */
+public class EnumMethodsGenerator {
+    private static final String VALUE_BY_TRANSFERABLE_ID_FIELD_NAME = "VALUE_BY_TRANSFERABLE_ID";
+
+    private static final String TRANSFERABLE_ID_FIELD_NAME = "transferableId";
+
+    /** Name of static method to get {@link Enum} by {@link TransferableEnum#transferableId}. */
+    public static final String FORM_TRANSFERABLE_ID_METHOD_NAME = "formTransferableId";
+
+    /** Postfix the name of the utility class for the {@link Enum} with {@link TransferableEnum}. */
+    public static final String TRANSFERABLE_UTILS_CLASS_POSTFIX = "TransferableUtils";
+
+    private final TypeUtils typeUtils;
+
+    /** Constructor. */
+    EnumMethodsGenerator(ProcessingEnvironment processingEnv) {
+        typeUtils = new TypeUtils(processingEnv);
+    }
+
+    /**
+     * Collects all {@link Enum}s that are present in network messages without the {@link Transient}.
+     *
+     * @param messageClasses Network message wrappers.
+     */
+    Set<TypeMirror> collectEnums(Collection<MessageClass> messageClasses) {
+        return messageClasses.stream()
+                .flatMap(messageClass -> messageClass.getters().stream())
+                .filter(element -> element.getAnnotation(Transient.class) == null)
+                .map(ExecutableElement::getReturnType)
+                .filter(this::isEnum)
+                .peek(this::checkEnumImplementTransferableEnum)
+                .collect(toSet());
+    }
+
+    /**
+     * Generates a utility class for the {@link Enum} with {@link TransferableEnum}.
+     *
+     * @param enumType {@link Enum} type mirror.
+     * @return Generated class like {@code SomeEnumTransferableUtils}.
+     */
+    TypeSpec generateEnumTransferableUtils(TypeMirror enumType) {
+        assert isEnum(enumType) && isTransferableEnum(enumType) : enumType;
+
+        TypeSpec.Builder typeSpecBuilder = TypeSpec.classBuilder(simpleEnumName(enumType) + "TransferableUtils")
```

Comment:
Will this create the class in the default package?

Reply:
Yep, adding a package happens here: `org.apache.ignite.internal.network.processor.TransferableObjectProcessor#generateEnumTransferableUtils`